src/insi-client.test.ts

```diff
@@ -362,137 +427,40 @@ describe('INSi Client', () => {
-    });
-
-    test('should respond with a CR01 code when the person is in the CR01 special case', async () => {
-      const person = new INSiPerson({
-        birthName: 'TCHITCHI',
-        firstName: 'OLA',
-        gender: Gender.Female,
-        dateOfBirth: '1936-06-21',
-      });
-
-      const {
-        requestId,
-        body,
-        rawBody,
-        bodyAsXMl,
-        requestBodyAsXML,
-      } = await insiClient.fetchIns(person, {
-        requestId: 'b3549edd-4ae9-472a-b26f-fd2fb4ef397f'
-      });
-
-      const expectedResponseAsXML = fs.readFileSync('src/fixtures/REP_CR01.xml', 'utf-8');
-
-      expect(requestId).toEqual('b3549edd-4ae9-472a-b26f-fd2fb4ef397f');
-      expect(body).toEqual(null);
-      expect(rawBody).toEqual({
-        CR: { CodeCR: '01', LibelleCR: 'Aucune identite trouvee' },
-      });
-      expect(bodyAsXMl).toEqual(expectedResponseAsXML);
-      expect(requestBodyAsXML).toEqual(getCNDAValidationXmlRequest({
+      const [response] = await insiClient.fetchIns(person, { requestId });
+      const requestBodyAsXML = getCNDAValidationXmlRequest({
         idam: IDAM,
         version: SOFTWARE_VERSION,
         name: SOFTWARE_NAME,
-        birthName: 'TCHITCHI',
-        firstName: 'OLA',
+        birthName: 'ADRTROIS-DOES-NOT-EXIST',
+        firstName: 'DOMINIQUE',
         sexe: Gender.Female,
-        dateOfBirth: '1936-06-21',
-      }));
+        dateOfBirth: '1997-02-26',
+      })
+
+      expect(response).toEqual({
+        status: INSiServiceRequestStatus.FAIL,
+        error: {
+          desirCode: 'insi_102',
+          error: 'L appel au service de recherche avec les traits d identité renvoie une erreur technique.',
+          siramCode: 'siram_40',
+          text: 'Le service est temporairement inaccessible.\nVeuillez renouveler votre demande ultérieurement. Si le problème persiste, contactez l\'éditeur du progiciel ou votre responsable informatique.',
+        },
+        requestBodyAsXML,
+        requestId,
+        response: null,
+        responseBodyAsJson: null,
+        responseBodyAsXml: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<soap:Envelope xmlns:soap=\"http://www.w3.org/2003/05/soap-envelope\"><env:Body xmlns:S=\"http://www.w3.org/2003/05/soap-envelope\" xmlns:env=\"http://www.w3.org/2003/05/soap-envelope\"><S:Fault xmlns:ns4=\"http://schemas.xmlsoap.org/soap/envelope/\"><S:Code><S:Value>S:Receiver</S:Value><S:Subcode><S:Value>S:siram_40</S:Value></S:Subcode></S:Code><S:Reason><S:Text xml:lang=\"en\">Le service est temporairement inaccessible.\nVeuillez renouveler votre demande ultérieurement. Si le problème persiste, contactez l'éditeur du progiciel ou votre responsable informatique.</S:Text></S:Reason><S:Detail><siram:Erreur severite=\"fatal\" code=\"insi_102\" xmlns:siram=\"urn:siram\">L appel au service de recherche avec les traits d identité renvoie une erreur technique.</siram:Erreur></S:Detail></S:Fault></env:Body></soap:Envelope>",
+      });
```

Comment:
pretty big file, it would love to be split into multiple files

Reply:
I will check this solution when i'm done with front and back.